bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/Unpacker.scala

```diff
@@ -36,35 +38,41 @@ case class Unpacker(s3Uploader: S3Uploader)(implicit s3Client: AmazonS3,
     val futureSummary = for {
       packageInputStream <- srcLocation.toInputStream
 
-      result <- Archive
+      unpackSummary <- Archive
         .unpack[UnpackSummary](packageInputStream)(unpackSummary) {
           (summary: UnpackSummary,
            inputStream: InputStream,
            archiveEntry: ArchiveEntry) =>
             if (!archiveEntry.isDirectory) {
-
-              val archiveEntrySize = putObject(
-                inputStream,
-                archiveEntry,
-                dstLocation
-              )
-
-              summary.copy(
-                fileCount = summary.fileCount + 1,
-                bytesUnpacked = summary.bytesUnpacked + archiveEntrySize
-              )
+              try {
+                val archiveEntrySize = putObject(
+                  inputStream,
+                  archiveEntry,
+                  dstLocation
+                )
+                summary.copy(
+                  fileCount = summary.fileCount + 1,
+                  bytesUnpacked = summary.bytesUnpacked + archiveEntrySize
+                )
+              } catch {
+                case ae: AmazonS3Exception =>
+                  throw new PutObjectLocationException(
+                    dstLocation,
+                    "upload failed",
```

Comment:
This is the sort of exception I think we should be keeping private – how the storage service manages itself is an internal concern. If somebody sees “upload failed” in an unpacker message, how do they fix that?

Reply:
Yup, looking again I think these exceptions should be specific to unpacking.